### ISSUE TITLE
`run-all show -json` mangles output for very large plans

### ISSUE THREAD

**Issue by hojerst** [gruntwork-io/terragrunt#3427]
## Describe the bug

When running `terragrunt run-all show -json` for very large plans, the output is mangled and no longer a valid json document. i assume its an issue with a line buffer, as the json from `terraform show -json` is all printed to a single line. using the normal, colored output the json document is cut after 65536 characters and when using `--terragrunt-forward-tf-stdout` the output seems to be cut at the beginning.

## Steps To Reproduce

1. create a plan which creates a json output of more than 65536 characters when running `terraform show -json planfile`
2. run `terragrunt run-all show -json` for the folder containing the plan

## Expected behavior

the json output should be complete without any content cut

## Nice to haves

- [ ] Terminal output
- [ ] Screenshots

## Versions

- Terragrunt version: v0.67.3
- OpenTofu/Terraform version: OpenTofu v1.8.2
- Environment details (Ubuntu 20.04, Windows 10, etc.): darwin_arm64

**Comment by denis256** [comment]
Hi,
I tried to set an example that render long output as json in https://github.com/denis256/terragrunt-tests/tree/master/show-json but output looks valid 

Can you share an example code/repo where this issue happens?

**Comment by hojerst** [comment]
i was able to reproduce the error in your test project with a slight change:

```bash
cp -a app app2
cp -a app app3
terragrunt run-all plan -out planfile
terragrunt run-all show -json planfile >out.json
```

looking at out.json, you will get a bunch of garbled json starting from line 2

**Comment by levkohimins** [comment]
Hi @hojerst, I repeated this

> cp -a app app2
cp -a app app3
terragrunt run-all plan -out planfile
terragrunt run-all show -json planfile >out.json

and what I see in `out.json` 

![1  tmuxinator local 2024-09-25 at 6 32 37 PM](https://github.com/user-attachments/assets/e96acca6-bbe0-45ab-991f-c68e0be7aa4a)

Could you show me what exactly is wrong here?

**Comment by hojerst** [comment]
[Uploading out.json…]()

**Comment by hojerst** [comment]
[out.json.gz](https://github.com/user-attachments/files/17161222/out.json.gz)

**Comment by levkohimins** [comment]
@hojerst 

Try with `--terragrunt-log-disable`, I guess it's what you need
```
terragrunt run-all show --terragrunt-log-disable -json planfile | sed -n 2p
```

**Comment by hojerst** [comment]
hm my version is 3 weeks old (0.67.3) and doesn't have this flag yet. i'm currently using nixpkgs for my toolchain and the latest version right now is 0.67.3... i'll try with the latest binary from here...

**Comment by hojerst** [comment]
i just ran the snippet multiple times, piping directly to jq, so i don't have to scroll and look for a broken json manually. the problem still persists with randomly broken json documents:

this is the output of two consecutive runs - the first printing the first json document fine, then failing on the second. the next call failed even on the first document / plan json output

```bash
[... snip - everything fine ...]
      ]
    }
  ],
  "timestamp": "2024-09-24T20:22:17Z",
  "errored": false
}
jq: parse error: Invalid literal at line 2, column 32785
❯ ~/.local/bin/terragrunt run-all show --terragrunt-log-disable -json planfile | jq
jq: parse error: Invalid literal at line 1, column 98321
```

**Comment by levkohimins** [comment]
I figured out why this happens. When we run `run-all show`, modules are executed in parallel, and the output from different modules starts to overlap, thus corrupting each other's results. So we need to buffer the output of each module, and then output it atomically. I'll fix this soon. @hojerst Thanks!

**Comment by levkohimins** [comment]
While we are working on a fix, you can use the `--terragrunt-parallelism 1` flag as a workaround.

**Comment by levkohimins** [comment]
Resolved in [v0.68.2-beta2024101501](https://github.com/gruntwork-io/terragrunt/releases/tag/v0.68.2-beta2024101501) release.
